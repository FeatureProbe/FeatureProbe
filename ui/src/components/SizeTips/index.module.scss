.size-tips {
  position: fixed;
  top: 60px;
  right: 20px;
  width: 348px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.02), 0 4px 6px 6px rgba(0, 0, 0, 0.02), 0 4px 6px 0 rgba(0, 0, 0, 0.06);
  border-radius: 12px;
  padding: 16px 24px;
  z-index: 9999;

  & > .main {
    display: flex;

    & > .icon {
      margin-right: 8px;

      & > .warning-icon {
        font-size: 20px;
        color: #ffc300;
        vertical-align: text-top;
      }
    }

    & > .content {
      & > .tips-title {
        font-size: 16px;
        color: #000000de;
        letter-spacing: 0;
        line-height: 24px;
        font-weight: 500;
        margin-bottom: 8px;
      }

      & > .tips-content {
        font-size: 14px;
        color: rgba(0, 0, 0, 0.6);
        letter-spacing: 0;
        font-weight: 400;
      }
    }
  }

  & > .actions {
    display: flex;
    justify-content: flex-end;
    margin-top: 24px;

    & > .button {
      display: inline-block;
      height: 24px;
      min-height: 0;
      font-size: 12px;
      color: #fff;
      letter-spacing: 0;
      line-height: 24px;
      font-weight: 400;
      white-space: nowrap;
    }
  }
}