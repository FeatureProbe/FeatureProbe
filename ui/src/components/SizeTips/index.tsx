import Icon from 'components/Icon';
import { FormattedMessage } from 'react-intl';
import { Button } from 'semantic-ui-react';
import styles from './index.module.scss';

interface SizeTipsProps {
  hide: boolean;
  onConfirm: () => void;
  size: number;
}

const SizeTips: React.FC<SizeTipsProps> = (props) => {
  const { hide, onConfirm, size } = props;

  return (
    <div hidden={hide} className={styles['size-tips']}>
      <div className={styles['main']}>
        <div className={styles['icon']}>
          <Icon type="warning-circle" customclass={styles['warning-icon']} />
        </div>
        <div className={styles['content']}>
          <div className={styles['tips-title']}>
            <FormattedMessage id="targeting.size.tips.title" />
          </div>
          <div className={styles['tips-content']}>
            <FormattedMessage id="targeting.size.tips" values={{ size: Math.floor(size / 1024) }} />
          </div>
        </div>
      </div>
      <div className={styles['actions']}>
        <Button onClick={onConfirm} className={styles['button']} primary>
          <FormattedMessage id="common.know.text" />
        </Button>
      </div>
    </div>
  );
};

export default SizeTips;
