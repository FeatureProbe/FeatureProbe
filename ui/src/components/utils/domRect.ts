export class DOMRect {
    x: number;
    width: number;
    y: number;
    height: number;
    constructor(x: number, y: number, width: number, height: number) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}