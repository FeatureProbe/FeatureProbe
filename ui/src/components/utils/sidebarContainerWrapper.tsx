import { SidebarContainer } from 'layout/hooks';

export default SidebarContainer.Provider;
