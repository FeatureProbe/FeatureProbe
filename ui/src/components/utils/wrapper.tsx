import en_US from '../../locales/en-US.json';
import { IntlProvider } from 'react-intl';

export const IntlWrapper: React.FC = ({ children }) => {
  return (
    <IntlProvider locale="en" messages={en_US}>{children}</IntlProvider>
  );
};
