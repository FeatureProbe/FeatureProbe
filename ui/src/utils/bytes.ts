export function getBytes(str: string) {
  let bytes = str.length,
    i = 0;
  for (; i < bytes; i++) if (str.charCodeAt(i) > 255) bytes++;

  return bytes;
}

const MB = 1048576;
export const MAX_SIZE = MB;