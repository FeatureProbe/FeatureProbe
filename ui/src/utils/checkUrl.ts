function escapeStringRegexp(string: string): string {
  if (typeof string !== 'string') {
    throw new TypeError('Expected a string');
  }

  return string
    .replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
    .replace(/-/g, '\\x2d');
}

export function matchUrl(matcher: string, originUrl: string, targetUrl: string): boolean {
  let href = '';
  let hash = '';
  let search = '';

  try {
    const fullUrl = new URL(targetUrl, '');
    href = fullUrl.href;
    hash = fullUrl.hash;
    search = fullUrl.search;
  } catch (error) {
    href = targetUrl;
  }

  let regex;
  let testUrl;

  switch (matcher) {
    case 'EXACT':
      testUrl = href;
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SIMPLE':
      testUrl = href.replace(hash, '').replace(search, '');
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SUBSTRING':
      testUrl = href.replace(search, '');
      regex = new RegExp('.*' + escapeStringRegexp(originUrl) + '.*$');
      break;
    case 'REGULAR':
      testUrl = href.replace(search, '');
      regex = new RegExp(originUrl);
      break;
    default:
      return false;
  }

  return regex.test(testUrl);
}