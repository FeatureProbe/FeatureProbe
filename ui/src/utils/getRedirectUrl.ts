import { getProjectList } from 'services/project';
import { IProject, IEnvironment } from 'interfaces/project';
import { IDictionary } from 'interfaces/targeting';
import { getFromDictionary } from 'services/dictionary';
import { LAST_SEEN } from 'constants/dictionary_keys';

const getInfo = async () => {
  const res =  await getFromDictionary<IDictionary>(LAST_SEEN);
  const { success, data } = res;
  if (success && data) {
    return JSON.parse(data.value);
  } else {
    return {};
  }
};

export const getRedirectUrl = async (defaultRedirectUrl: string) => {
  let redirectUrl = defaultRedirectUrl;
  const info = await getInfo();
  const { projectKey, environmentKey } = info;

  if (!!projectKey && !!environmentKey) {
    redirectUrl = `/${projectKey}/${environmentKey}/toggles`;
  } else {
    const res = await getProjectList<IProject[]>();
    const { success, data } = res;
    if (success && data) {
      if (data[0]) {
        const { key, environments } = data[0];
        let envKey = '';
        environments.some((environment: IEnvironment) => {
          if (environment.key) {
            envKey = environment.key;
            return true;
          }
          return environment;
        });
  
        if (key && envKey) {
          redirectUrl = `/${key}/${envKey}/toggles`;
        }
      } else {
        redirectUrl = '/projects';
      }
    }
  }

  return redirectUrl;
};