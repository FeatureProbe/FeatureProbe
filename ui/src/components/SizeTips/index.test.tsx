import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import SizeTips from '.';
import { IntlWrapper } from 'components/utils/wrapper';

it('SizeTips snapshot', (done) => {
  (async () => {
    const { asFragment, rerender } = render(
      <SizeTips
        hide={false}
        onConfirm={function (): void {
          throw new Error('Function not implemented.');
        }}
        size={0}
      />,
      {
        wrapper: IntlWrapper,
      }
    );
    expect(asFragment()).toMatchSnapshot();

    rerender(
      <SizeTips
        hide={true}
        onConfirm={function (): void {
          throw new Error('Function not implemented.');
        }}
        size={0}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    done();
  })();
});

test('click SizeTips', (done) => {
  (async () => {
    const handleConfirm = jest.fn();
    render(<SizeTips onConfirm={handleConfirm} size={0} hide={false}></SizeTips>, {
      wrapper: IntlWrapper,
    });

    await userEvent.click(screen.getByText('I see'));
    expect(handleConfirm).toBeCalled();

    done();
  })();
});
