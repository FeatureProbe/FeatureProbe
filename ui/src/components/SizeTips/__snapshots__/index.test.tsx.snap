// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeTips snapshot 1`] = `
<DocumentFragment>
  <div
    class="size-tips"
  >
    <div
      class="main"
    >
      <div
        class="icon"
      >
        <span
          class="iconfont icon-warning-circle warning-icon"
          customclass="warning-icon"
          type="warning-circle"
        />
      </div>
      <div
        class="content"
      >
        <div
          class="tips-title"
        >
          Warning tips
        </div>
        <div
          class="tips-content"
        >
          The total content of this page can not exceed 1 MB, and will not be released normally when it is larger than 1 MB. Currently, it has reached 0 KB!
        </div>
      </div>
    </div>
    <div
      class="actions"
    >
      <button
        class="ui primary button button"
      >
        I see
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SizeTips snapshot 2`] = `
<DocumentFragment>
  <div
    class="size-tips"
    hidden=""
  >
    <div
      class="main"
    >
      <div
        class="icon"
      >
        <span
          class="iconfont icon-warning-circle warning-icon"
          customclass="warning-icon"
          type="warning-circle"
        />
      </div>
      <div
        class="content"
      >
        <div
          class="tips-title"
        >
          Warning tips
        </div>
        <div
          class="tips-content"
        >
          The total content of this page can not exceed 1 MB, and will not be released normally when it is larger than 1 MB. Currently, it has reached 0 KB!
        </div>
      </div>
    </div>
    <div
      class="actions"
    >
      <button
        class="ui primary button button"
      >
        I see
      </button>
    </div>
  </div>
</DocumentFragment>
`;
